File: 이훈일/Programmers/Level2/예상 대진표.java
class Solution {
    public int solution(int n, int a, int b) {
        int answer = 0;
        int start = 0;
        
        while(true) {
            int compare = n/2 + start;
            if(a <= compare && b <= compare) {
                n = n / 2;
            } else if(a > compare && b > compare) {
                start = compare;
                n = n / 2;
            } else {
                break;
            }
        }
        
        for(int i = 1; i <= 20; i++) {
            if(n == Math.pow(2, i)) {
                answer = i;
                break;
            }
        }

        return answer;
    }
}

File: 조준장/level2/예상 대진표.kt
class Solution {
    fun solution(n: Int, a: Int, b: Int): Int {
        var answer = 0
        var new_a : Int = a
        var new_b : Int = b
        while (new_a != new_b){
            answer += 1
            new_a = (new_a + 1) / 2
            new_b = (new_b + 1) / 2
        }
        
        return answer
    }
}
